modules/REST/V1/Languages.php

```diff
@@ -0,0 +1,809 @@
+				array( 'status' => 400 )
+			);
+		}
+
+		/**
+		 * $request: locale, slug, name, is_rtl, term_group, flag_code
+		 * $args:    locale, slug, name, rtl,    term_group, flag? (from add_language())
+		 */
+		if ( ! isset( $request['slug'], $request['name'], $request['is_rtl'], $request['flag_code'] ) ) {
+			$languages = include POLYLANG_DIR . '/settings/languages.php';
+
+			if ( empty( $languages[ $request['locale'] ] ) ) {
+				return new WP_Error(
+					'rest_invalid_locale',
+					__( 'The locale is invalid.', 'polylang' ),
+					array( 'status' => 400 )
+				);
+			}
+
+			$language = $languages[ $request['locale'] ];
+			$defaults = array(
+				'slug'       => $language['code'],
+				'name'       => $language['name'],
+				'rtl'        => 'rtl' === $language['dir'],
+				'flag'       => $language['flag'],
+				'term_group' => 0,
+			);
+		} else {
+			$defaults = array(
+				'term_group' => 0,
+			);
+		}
+
+		$prepared = (array) $this->prepare_item_for_database( $request );
+		$prepared = array_merge( $defaults, $prepared );
+
+		/**
+		 * @phpstan-var array{
+		 *     locale: non-empty-string,
+		 *     slug: non-empty-string,
+		 *     name: non-empty-string,
+		 *     rtl: bool,
+		 *     term_group: int,
+		 *     flag: non-empty-string,
+		 *     no_default_cat?: bool
+		 * } $prepared
+		 */
+		$result = $this->languages->add( $prepared );
+
+		if ( is_wp_error( $result ) ) {
+			return $this->add_status_to_error( $result );
+		}
+
+		/** @var PLL_Language */
+		$language = $this->languages->get( $prepared['slug'] );
+		return $this->prepare_item_for_response( $language, $request );
+	}
+
+	/**
+	 * Retrieves one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array{
+	 *     term_id: int
+	 * }|array{
+	 *     slug: non-empty-string
+	 * }
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_item( $request ) {
+		$language = $this->get_language( $request );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
+
+		return $this->prepare_item_for_response( $language, $request );
+	}
+
+	/**
+	 * Updates one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array{
+	 *     term_id: int,
+	 *     slug?: non-empty-string,
+	 *     locale?: non-empty-string,
+	 *     name?: non-empty-string,
+	 *     is_rtl?: bool,
+	 *     term_group?: int,
+	 *     flag_code?: non-empty-string
+	 * }
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function update_item( $request ) {
+		$language = $this->get_language( $request );
+
+		if ( is_wp_error( $language ) ) {
+			return $language;
+		}
```

Comment:
Shouldn't we call `add_status_to_error()`?

Reply:
Same as above.